live-build/misc/migration-scripts/dx_execute

```diff
@@ -97,6 +97,21 @@ LX_RDS="$LX_RDS_PARENT/root"
 	die "the expected Linux RDS ($LX_RDS) was either not found or has" \
 		"been specified more than once in the bootloader's menu file"
 
+#
+# Replace the empty /var/delphix dataset from the migration image by a clone
+# of the current /var/delphix dataset.
+#
+LX_VAR_DLPX="$LX_RDS_PARENT/data"
+LX_CONTAINER="${LX_RDS_PARENT##*/}"
+CUR_VAR_DLPX=$(mount | awk '/^\/var\/delphix /{ print $3 }')
+[[ -n $CUR_VAR_DLPX ]] || die "could not determine current /var/delphix dataset"
+zfs list "$CUR_VAR_DLPX" >/dev/null 2>&1 ||
+	die "'$CUR_VAR_DLPX' is not a valid zfs dataset"
+zfs destroy "$LX_VAR_DLPX" || die "failed to destroy '$LX_VAR_DLPX'"
+zfs snapshot "${CUR_VAR_DLPX}@${LX_CONTAINER}"
+zfs clone -o compression=on -o mountpoint=legacy \
```

Comment:
Will `rootfs-container delete` still work correctly on the initial dataset used for the LX migration? Does it currently work without this change?

Right now we can do a not-in-place upgrade, then use `rootfs-container delete` to remove the old rootfs container/datasets. So, I'm wondering if we'll be able to do a migration, then a not-in-place upgrade, and still use the `rootfs-container delete` script to delete that dataset that was used for the initial migration?

As a tangential concern, "leaking" things from illumos to linux worries me a little bit since it can make it more difficult to mimic customer systems in-house. E.g. in this particular example, I could envision us testing the linux equivalent of `dx_delete` in-house, without ever taking migration into account. This could give us false positives when it works with migration out of the picture, but it breaks when the customer previously did a migration.

I don't mean to imply this is wrong, I just think we need to be mindful about all the possible implications that our migration implementation might have on other parts of our software.

Reply:
We'll definitely need to do some dataset cleanup after rebooting into Linux like promoting the clone and deleting the old Illumos datasets.

I do understand your concerns about "leaking" config from the migration process, however this is bound to happen in all the aspects of the product (i.e. datasets on domain0 that were created earlier or some OS configurations). We should strive to reduce those differences where we can, and just be mindful of doing the proper testing where we cannot. We do have plans of testing Linux upgrade after migration.